cmd/lifecycle/creator.go

```diff
@@ -260,12 +274,35 @@ func (c *createCmd) Exec() error {
 
 func (c *createCmd) registryImages() []string {
 	var registryImages []string
-	if c.cacheImageTag != "" {
-		registryImages = append(registryImages, c.cacheImageTag)
+	if c.cacheImageRef != "" {
+		registryImages = append(registryImages, c.cacheImageRef)
 	}
 	if !c.useDaemon {
-		registryImages = append(registryImages, append([]string{c.imageName}, c.additionalTags...)...)
-		registryImages = append(registryImages, c.runImageRef, c.previousImage)
+		registryImages = append(registryImages, append([]string{c.outputImageRef}, c.additionalTags...)...)
+		registryImages = append(registryImages, c.runImageRef, c.previousImageRef)
 	}
 	return registryImages
 }
+
+func (c *createCmd) platformAPIVersionGreaterThan06() bool {
+	return api.MustParse(c.platform.API()).Compare(api.MustParse("0.7")) >= 0
+}
+
+func (c *createCmd) ReadableImages() []string {
+	var readableImages []string
+	if c.platformAPIVersionGreaterThan06() {
+		if !c.useDaemon {
+			readableImages = append(readableImages, c.previousImageRef, c.runImageRef)
+		}
+	}
+	return readableImages
+}
+func (c *createCmd) WriteableImages() []string {
+	var writeableImages []string
+	if c.platformAPIVersionGreaterThan06() {
+		if !c.useDaemon {
+			writeableImages = append(writeableImages, append([]string{c.cacheImageRef}, c.additionalTags...)...)
```

Comment:
Same comment as https://github.com/buildpacks/lifecycle/pull/646/files#r666182517

Reply:
Fixed in the following [commit](https://github.com/buildpacks/lifecycle/pull/646/commits/5c900400a2e89ae0d9f8a8d0c02243dec574ceba)